unittests/AST/ASTImporterTest.cpp

```diff
@@ -1996,6 +1996,193 @@ TEST_P(ASTImporterOptionSpecificTestBase, ReimportWithUsedFlag) {
   EXPECT_TRUE(Imported2->isUsed(false));
 }
 
+// FIXME Can we unify these tests with ImportClassTemplates?
+struct ImportFunctionTemplates : ASTImporterOptionSpecificTestBase {};
+
+TEST_P(ImportFunctionTemplates,
+       PrototypeShouldBeImportedAsAPrototypeWhenThereIsNoDefinition) {
+  Decl *FromTU = getTuDecl("template <class T> void X();", Lang_CXX);
+  auto Pattern = functionTemplateDecl(hasName("X"), unless(isImplicit()));
+  auto FromD = FirstDeclMatcher<FunctionTemplateDecl>().match(FromTU, Pattern);
```

Comment:
I do not know why but `auto *` is the usual way of this kind of declarations (in this file).